torchdynamo/_guards.cpp

```diff
@@ -99,11 +131,21 @@ class TensorCheck {
       const auto &strides = v.strides();
       for (size_t i = 0; i < ndim; ++i) {
         if (sizes_[i] != sizes[i]) {
-          // return fmt::format("tensor size mismatch at index {}. expected {},
-          // actual {}", i, sizes_[i], sizes[i]);
-          fail_reason << "size mismatch at index " << i << ". expected "
-                      << sizes_[i] << ", actual " << sizes[i];
-          return fail_reason.str();
+          bool is_dynamic_dim = false;
+          for (int dd = 0; dd < dynamic_dims_.size(); dd++) {
```

Comment:
I am seeing a lot of duplicate code, we should refactor this to reduce duplication.

Reply:
Thats the nature of this whole file, `check_verbose` and `check` are the same logic with marginally different return types - this is probably outside the scope of this PR.